oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2AuthorizationServerMetadataEndpointFilter.java

```diff
@@ -83,6 +83,7 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 				.tokenEndpoint(asUrl(this.providerSettings.getIssuer(), this.providerSettings.getTokenEndpoint()))
 				.tokenEndpointAuthenticationMethods(clientAuthenticationMethods())
 				.jwkSetUrl(asUrl(this.providerSettings.getIssuer(), this.providerSettings.getJwkSetEndpoint()))
+				.userInfoEndpoint(asUrl(this.providerSettings.getIssuer(), this.providerSettings.getOidcUserInfoEndpoint()))
```

Comment:
The UserInfo endpoint is not defined in RFC 8414 [2.  Authorization Server Metadata](https://datatracker.ietf.org/doc/html/rfc8414#section-2) and instead defined only in [3. OpenID Provider Metadata](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata).

Therefore, the update should only be accessible in `OidcProviderConfiguration` not in `OAuth2AuthorizationServerMetadata`

Reply:
Hi @jgrandja ... Got it thanks : ) I added it because of the comment at the end of Section [2.](https://datatracker.ietf.org/doc/html/rfc8414#section-2) "Additional authorization server metadata parameters MAY also be used" and the sample in section [3.2. Authorization Server Metadata Response](https://datatracker.ietf.org/doc/html/rfc8414#section-3.2) shows userinfo_endpoint. But more so, I added it because I wasn't sure if the ClientRegistration within oauth2-client would be configured using the openid configuration endpoint or an authorization server metadata endpoint. After looking into the docs, it appears it tries both. So your call ; ) Let me know.